eos-install-firefox

```diff
@@ -0,0 +1,354 @@
+#!/bin/bash -e
+
+# Downloads Firefox and unzips to the user's home directory,
+# with a desktop launcher
+
+exit_with_error() {
+   echo $1
+   exit 1
+}
+
+USERID=$(id -u)
+if [ "$USERID" == "0" ]; then
+    exit_with_error "Program cannot be run as superuser"
+fi
+
+DEB_ARCH=`dpkg --print-architecture`
+case ${DEB_ARCH} in
+    amd64) ;; # Only 64-bit Intel architectures are supported
+    *) exit_with_error "Unsupported architecture ${DEB_ARCH}"
+       ;;
+esac
+
+TEMP_DIR=`mktemp -d`
+DEST_DIR=~/.local/share/eos-third-party-apps/firefox
+WGET_COMMAND="wget -c --tries=5"
+DOWNLOAD_DIR=`xdg-user-dir DOWNLOAD`
+
+# List of supported Firefox languages, from https://download-installer.cdn.mozilla.net/pub/firefox/releases/latest/README.txt
+FIREFOX_LANGS="ach af an ar as ast az be bg bn-BD bn-IN br bs ca cs cy da de dsb el en-GB en-US en-ZA eo es-AR es-CL es-ES es-MX et eu fa ff fi fr fy-NL ga-IE gd gl gu-IN he hi-IN hr hsb hu hy-AM id is it kk km kn ko lij lt lv mai mk ml mr ms nb-NO nl nn-NO or pa-IN pl pt-BR pt-PT rm ro ru si sk sl son sq sr sv-SE ta te th tr uk uz vi xh zh-CN zh-TW"
+
+JAVA_BUILD=8u112-b15
+
+# While the following would be a faster download of the current version,
+# previous versions are often removed from the server, so it is safest
+# for an automated script to use the stable version in the archive
+# http://fpdownload.macromedia.com/get/flashplayer/pdc/11.2.202.643/install_flash_player_11_linux.x86_64.tar.gz
+FLASH_VERSION=11.2.202.643
+FLASH_SHA256SUM="1a79f38f7153d436083b5a6b5d0f38e0c4cd345e83e7d050152452057f83350a"
+FLASH_FILENAME=fp_${FLASH_VERSION}_archive.zip
+FLASH_URL=http://download.macromedia.com/pub/flashplayer/installers/archive/${FLASH_FILENAME}
+
+# Find the closest supported Firefox for the user's current language
+locale=`echo ${LANG} | cut -f 1 -d . | sed s/_/-/`
+lang=`echo ${locale} | cut -f 1 -d -`
+firefox_lang=
+for l in ${FIREFOX_LANGS}
+do
```

Comment:
Nit: I think we normally put `while` and `do` in the same line, as in `while COND; do`. This applies to other cases after this point.